File: snippets/python/src/service_status.py
import breez_sdk

def health_check_status():
    try: 
        # ANCHOR: health-check-status
        health_check = breez_sdk.service_health_check(api_key="<api key>")
        print("Current service status is: ", health_check.status)
        # ANCHOR_END: health-check-status
    except Exception as error: 
        print(error)
        raise

def report_payment_failure(sdk_services):
    try:
        # ANCHOR: report-payment-failure
        payment_hash = "..."
        req = breez_sdk.ReportIssueRequest.PAYMENT_FAILURE(
            data=breez_sdk.ReportPaymentFailureDetails(
                payment_hash=payment_hash))
        sdk_services.report_issue(req=req)
        # ANCHOR_END: report-payment-failure
    except Exception as error:
        print(error)
        raise

File: snippets/rust/src/service_status.rs
use std::sync::Arc;

use anyhow::Result;
use breez_sdk_core::*;
use log::info;

async fn health_check_status(sdk: Arc<BreezServices>) -> Result<()> {
    // ANCHOR: health-check-status
    let api_key = "<api key>".to_string();

    let health_check = BreezServices::service_health_check(api_key).await?;
    info!("Current service status is: {:?}", health_check.status);
    // ANCHOR_END: health-check-status

    Ok(())
}

async fn report_payment_failure(sdk: Arc<BreezServices>) -> Result<()> {
    // ANCHOR: report-payment-failure
    let payment_hash = String::from("...");

    sdk.report_issue(ReportIssueRequest::PaymentFailure {
        data: ReportPaymentFailureDetails {
            payment_hash,
            comment: None,
        },
    })
    .await?;
    // ANCHOR_END: report-payment-failure

    Ok(())
}
